jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jsonb/JsonbDiagnosticsCollector.java

```diff
@@ -87,21 +87,25 @@ private void collectJsonbTransientFieldDiagnostics(ICompilationUnit unit,
             List<Diagnostic> diagnostics) throws JavaModelException {
         for (IType type : unit.getAllTypes()) {
             for (IField field : type.getFields()) {
-                List<String> jsonbAnnotationsForField = getJsonbAnnotationNamesForField(field);
+                List<String> jsonbAnnotationsForField = getJsonbAnnotationNames(field);
                 if (jsonbAnnotationsForField.contains(JsonbConstants.JSONB_TRANSIENT)) {
-                    if (hasJsonbAnnotationOtherThanTransient(field)) {
-                        Diagnostic diagnostic = createDiagnosticBy(unit, field, JsonbConstants.JSONB_TRANSIENT);
-                        diagnostic.setData((JsonArray)(new Gson().toJsonTree(jsonbAnnotationsForField)));
-                        diagnostics.add(diagnostic);
-                    }
+                    boolean hasAccessorConflict = false;
                     // Diagnostics on the getter and setter of the field are created when they are
                     // annotated with Jsonb annotations other than JsonbTransient.
                     for (IMethod accessor : JDTUtils.getAccessors(field)) {
                         if (hasJsonbAnnotationOtherThanTransient(accessor)) {
+                            List<String> jsonbAnnotationsForAccessor = getJsonbAnnotationNames(accessor);
                             Diagnostic diagnostic = createDiagnosticBy(unit, accessor, JsonbConstants.JSONB_TRANSIENT);
+                            diagnostic.setData((JsonArray)(new Gson().toJsonTree(jsonbAnnotationsForAccessor)));
                             diagnostics.add(diagnostic);
+                            hasAccessorConflict = true;
                         }
                     }
+                    if (hasJsonbAnnotationOtherThanTransient(field) || hasAccessorConflict) {
```

Comment:
Just a question for clarification:
if we have a scenario like this:
```java
@JsonbTransient
private int id;

@JsonbProperty
public int getId() {}
```
This will now create 2 diagnostics: one for the field and one for the accessor right?

Reply:
Yes, for the field the associated quick fix would be "Remove @JsonbTransient" and for the accessor it would be "Remove @JsonbProperty".